app/assets/javascripts/darkswarm/directives/on_hand.js.coffee

```diff
@@ -13,11 +13,16 @@ Darkswarm.directive "ofnOnHand", ->
       ngModel.$setDirty = setDirty
 
     ngModel.$parsers.push (viewValue) ->
-      on_hand = parseInt(attr.ofnOnHand)
-      if parseInt(viewValue) > on_hand
-        alert t("js.insufficient_stock", {on_hand: on_hand})
-        viewValue = on_hand
+      available_quantity = scope.available_quantity()
+      if parseInt(viewValue) > available_quantity
+        alert t("js.insufficient_stock", {on_hand: available_quantity})
+        viewValue = available_quantity
         ngModel.$setViewValue viewValue
         ngModel.$render()
 
       viewValue
+
+    scope.available_quantity = ->
+      on_hand = parseInt(attr.ofnOnHand)
+      finalized_quantity = parseInt(attr.finalizedquantity) || 0 # finalizedquantity is optional
+      on_hand + finalized_quantity
```

Comment:
IMO this deserves some unit testing

Reply:
yeah, agree but we only have one directive unit test. I gave it a try to create one for this now but it's not working... I remember I spent a huge amount of time in the past trying to test one directive without success...
and this one in particular using "ngModel.$setViewValue"...

Reply:
can we then create a tech debt issue with the problems you had :pray: ?

Reply:
ah, yes, great point, I'll do that.
I dont know exactly what were the problems I had but a tech debt issue for angular directive unit tests will be good!

Reply:
here: https://github.com/openfoodfoundation/openfoodnetwork/issues/3947